### ISSUE TITLE
Migrate XML updates from ;xmlpatch

### ISSUE THREAD

**Issue by sandersch** [elanthia-online/scripts#97]
# Migrate XML updates from ;xmlpatch

`;xmlpatch` was created a workaround for getting changes merged into the upstream `gameobj-data.xml`. Now that we own the `gameobj-data.xml` file published in the repo, we can create migrations for the published data and make the script obsolete. I am getting started on the work and wanted to have a discussion about some of the changes required so that we're all on the same page. There are two main types of XML updates in `;xmlpatch`:

### Updates to existing types

These are pretty straightforward and I've started creating migrations for these. I should have some PRs to send soon.

* Handful of additions and exclusions to existing types (jewelry, armor, clothing, etc)
* Creatures and gems from new areas (SoS, Reim)
* Adds optional prefix `shifting` to existing box names

### New Type Categories

These are all new types that don't currently exist in `gameobj-data.xml`. I think that they are all worthwhile additions that make sense for us to merge upstream, but if anyone feels like adding these may cause issues, please let me know.

* `reim` - creature and gem category.
* `moonshard` - Duskruin quest items
* `duskruin` - area category for gems
* `pennant chase` - Pennant Chase quest items
* `noncorporeal` -  creature category
* `companion` - new NPC category for ranger companions
* `familiar` - new NPC category for wizard familiars
* `phased` - new type for phased (704) boxes

**Comment by sandersch** [comment]
@ondreian @mrhoribu let me know if you see any issues with this plan especially the new type additions.

**Comment by sandersch** [comment]
The only `spell-list.xml` patch is the updated 909 messaging:
```ruby
def apply_spell_patches
    # Spell data patching
    Spell[909].instance_variable_set(:@msgup, "Faint ripples in the .* form beneath you for a moment\\.")
    Spell[909].instance_variable_set(:@msgdn, "Faint ripples in the .* beneath you become apparent before quickly dissipating\\.")
end
```
which has already been updated [upstream](https://github.com/matt-lowe/Lich/blob/master/data/spell-list.xml#L897).

**Comment by mrhoribu** [comment]
Btw, I added more of the RFD creatures for reim_npcs, not sure if best to put them here for you, but here was my modified reim_npcs with RFD creatures added in.

`ethereal (?:barbarian|knight|pillager|raider|scout|soldier|waylayer|zealot|commoner|denizen|guard|guardsman|guardswoman|inmate|lunatic|madman|madwoman|peasant|prisoner|squire|swordsman|swordswoman|traveller|townsman|townswoman|villager)|ghostly (?:bandit|highwayman|highwaywoman|marauder|waylayer)|unworldly (?:barbarian|knight|pillager|raider|scout|soldier|waylayer|zealot|guest|maid|noble|royal guard|royal knight|servant|slave|steward|visitor)|celestial (?:dancer|juggler|nomad|traveller)|Shopkeeper|Innkeeper|Bartender|Patrol Leader|Bandit Lord|Bandit Lady|Gypsy Queen|Gypsy King|Guard Captain|Wall Captain|Drill Sergeant|Stable Hostler|Dungeon Master|Master Torturer|Butler|Cook|Knight Captain|Foreign Dignitary|Royal Prince|Royal Princess|Royal Jester|Royal Emperor|Royal Empress|Sapper Lord|Sapper Lady|Raid Leader|Dark Knight|Massive Butcher|Slender Mercenary|Darkly Hooded Figure|Crimson Count|Kennel Master|Royal Inquisitor|Towering Crusader|Hulking Berserker|Jester Ser Rets|Dame Venti|Corrupted Knight Owrym|Arch Priest Dlitse|Dame Elle Zo|Thane Wedge|Dark Astrologer Scionae|speedy sapper`

**Comment by mrhoribu** [comment]
Reviewing this, I like it all. The `realm:area` and `event:name` namespacing makes perfect sense. I welcome the addition of the other new TYPES as well for `noncorporeal`, `companion`, `familiar` and `phased`. Once this is done I'm sure there are other types we can discuss to add as well. But namely need to.

The only consideration I would see is if there's really a need to specify between the difference of a `companion` vs a `familiar`. Not sure what a merged name would be or if it's better to keep them separate.

**Comment by sandersch** [comment]
I don't have a strong opinion about familiars vs companions. My initial thought was that we should match `;xmlpatch` as long as it makes sense to minimize issues with scripts that rely on `xmlpatch`'s changes, but  I think combining them would be fine too.

**Comment by sandersch** [comment]
Phase 1 (modifications to existing types) is complete. My plan is to add a `create_table(:type_name)` to the migration DSL so that we can add these programmatically.

**Comment by sandersch** [comment]
Copious data on companions in case that is helpful: http://drfuturepast.com/GSIV/GECP/
